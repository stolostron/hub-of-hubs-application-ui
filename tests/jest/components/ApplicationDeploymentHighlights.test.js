/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2016, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const React = require("react");
const renderer = require("react-test-renderer");
const ApplicationDeploymentHighlights = require("../../../src-web/components/ApplicationDeploymentHighlights")
  .default;

const applications = {
  items: [
    {
      _uid: "local-cluster/5cd1d4c7-52aa-11ea-bf05-00000a102d26",
      name: "mortgage-app",
      namespace: "default",
      dashboard:
        "https://localhost:443/grafana/dashboard/db/mortgage-app-dashboard-via-federated-prometheus?namespace=default",
      clusterCount: 1,
      hubSubscriptions: [
        {
          _uid: "local-cluster/5cdc0d8d-52aa-11ea-bf05-00000a102d26",
          status: "Propagated",
          channel: "default/mortgage-channel",
          __typename: "Subscription"
        }
      ],
      created: "2020-02-18T23:57:04Z",
      __typename: "Application"
    },
    {
      _uid: "local-cluster/42d4c013-52b9-11ea-bf05-00000a102d26",
      name: "samplebook-gbapp",
      namespace: "sample",
      dashboard:
        "https://localhost:443/grafana/dashboard/db/samplebook-gbapp-dashboard-via-federated-prometheus?namespace=sample",
      clusterCount: 1,
      hubSubscriptions: [
        {
          _uid: "local-cluster/42d9ec27-52b9-11ea-bf05-00000a102d26",
          status: "Propagated",
          channel: "gbook-ch/guestbook",
          __typename: "Subscription"
        }
      ],
      created: "2020-02-19T01:43:43Z",
      __typename: "Application"
    },
    {
      _uid: "local-cluster/8f4799db-4cf4-11ea-a229-00000a102d26",
      name: "stocktrader-app",
      namespace: "stock-trader",
      dashboard: null,
      clusterCount: 0,
      hubSubscriptions: [],
      created: "2020-02-11T17:33:04Z",
      __typename: "Application"
    },
    {
      _uid: "local-cluster/e77e69a7-4d25-11ea-a229-00000a102d26",
      name: "subscribed-guestbook-application",
      namespace: "kube-system",
      dashboard: null,
      clusterCount: 2,
      hubSubscriptions: [
        {
          _uid: "local-cluster/98dce449-52b8-11ea-bf05-00000a102d26",
          status: "Propagated",
          channel: "default/hub-local-helm-repo",
          __typename: "Subscription"
        }
      ],
      created: "2020-02-11T23:26:18Z",
      __typename: "Application"
    }
  ]
};

const header = {
  breadcrumbItems: [
    { url: "/multicloud/applications" },
    { url: "/multicloud/applications/default/mortgage-app" }
  ]
};

describe("ApplicationDeploymentHighlights with an app selected", () => {
  const mockStore = configureStore([]);
  let store;
  // values from the mocked store
  beforeEach(() => {
    store = mockStore({
      QueryApplicationList: applications,
      secondaryHeader: header
    });
  });

  it("ApplicationDeploymentHighlights renders correctly with an app selected.", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ApplicationDeploymentHighlights />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("ApplicationDeploymentHighlights with no app", () => {
  const mockStore = configureStore([]);
  let store;
  // values from the mocked store
  beforeEach(() => {
    store = mockStore({});
  });

  it("ApplicationDeploymentHighlights with no app renders correctly.", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ApplicationDeploymentHighlights />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("ApplicationDeploymentHighlights with no app selection", () => {
  const mockStore = configureStore([]);
  let store;
  // values from the mocked store
  beforeEach(() => {
    store = mockStore({
      QueryApplicationList: applications,
      secondaryHeader: { breadcrumbItems: undefined }
    });
  });
  it("ApplicationDeploymentHighlights renders correctly when no app is selected.", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ApplicationDeploymentHighlights />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
