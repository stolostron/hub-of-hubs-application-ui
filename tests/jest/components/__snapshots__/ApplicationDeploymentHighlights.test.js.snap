// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationDeploymentHighlights with an app selected ApplicationDeploymentHighlights renders correctly with an app selected. 1`] = `
<div
  id="DeploymentHighlights"
>
  <div
    id="ApplicationDeploymentHighlightsTerminology"
  >
    <article
      className="pf-c-card pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Card-1"
      data-ouia-component-type="PF4/Card"
      data-ouia-safe={true}
      id=""
    >
      <div
        className="pf-c-card__header makeStyles-header-1"
        onClick={[Function]}
      >
        <div
          className="pf-c-card__header-toggle"
        >
          <button
            aria-disabled={false}
            aria-expanded={true}
            aria-label="Toggle details"
            className="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={false}
            id="toggle-button"
            onClick={[Function]}
            role={null}
            type="button"
          >
            <span
              className="pf-c-card__header-toggle-icon"
            >
              <svg
                aria-hidden="true"
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="pf-c-card__title"
        >
          Learn more about the terminology
        </div>
      </div>
      <div
        className="pf-c-card__expandable-content makeStyles-body-2"
      >
        <div
          className="pf-c-card__body"
        >
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Subscriptions
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Subscriptions identify Kubernetes resources within channels (source repositories). Then, the subscription places the Kubernetes resources on the target clusters.
            </p>
          </div>
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Placement rules
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Placement rules define the target clusters where subscriptions are delivered. This is done by cluster name, cluster resource annotation(s), or cluster resource label(s).
            </p>
          </div>
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Channels
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Channels point to repositories where Kubernetes resources are stored, such as Git, Helm chart, or object storage repositories, or Namespaces on the local cluster. Channels support multiple subscriptions from multiple targets.
            </p>
          </div>
          <div
            className="deployment-highlights-terminology-docs"
          >
            <a
              href="https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.4/html/applications/managing-applications"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="deployment-highlights-terminology-docs-text"
              >
                View documentation
              </span>
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                className="details-item-link-icon"
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </article>
  </div>
</div>
`;

exports[`ApplicationDeploymentHighlights with no app ApplicationDeploymentHighlights with no app renders correctly. 1`] = `
<div
  id="DeploymentHighlights"
>
  <div
    id="ApplicationDeploymentHighlightsTerminology"
  >
    <article
      className="pf-c-card pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Card-2"
      data-ouia-component-type="PF4/Card"
      data-ouia-safe={true}
      id=""
    >
      <div
        className="pf-c-card__header makeStyles-header-1"
        onClick={[Function]}
      >
        <div
          className="pf-c-card__header-toggle"
        >
          <button
            aria-disabled={false}
            aria-expanded={true}
            aria-label="Toggle details"
            className="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={false}
            id="toggle-button"
            onClick={[Function]}
            role={null}
            type="button"
          >
            <span
              className="pf-c-card__header-toggle-icon"
            >
              <svg
                aria-hidden="true"
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="pf-c-card__title"
        >
          Learn more about the terminology
        </div>
      </div>
      <div
        className="pf-c-card__expandable-content makeStyles-body-2"
      >
        <div
          className="pf-c-card__body"
        >
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Subscriptions
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Subscriptions identify Kubernetes resources within channels (source repositories). Then, the subscription places the Kubernetes resources on the target clusters.
            </p>
          </div>
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Placement rules
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Placement rules define the target clusters where subscriptions are delivered. This is done by cluster name, cluster resource annotation(s), or cluster resource label(s).
            </p>
          </div>
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Channels
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Channels point to repositories where Kubernetes resources are stored, such as Git, Helm chart, or object storage repositories, or Namespaces on the local cluster. Channels support multiple subscriptions from multiple targets.
            </p>
          </div>
          <div
            className="deployment-highlights-terminology-docs"
          >
            <a
              href="https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.4/html/applications/managing-applications"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="deployment-highlights-terminology-docs-text"
              >
                View documentation
              </span>
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                className="details-item-link-icon"
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </article>
  </div>
</div>
`;

exports[`ApplicationDeploymentHighlights with no app selection ApplicationDeploymentHighlights renders correctly when no app is selected. 1`] = `
<div
  id="DeploymentHighlights"
>
  <div
    id="ApplicationDeploymentHighlightsTerminology"
  >
    <article
      className="pf-c-card pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Card-3"
      data-ouia-component-type="PF4/Card"
      data-ouia-safe={true}
      id=""
    >
      <div
        className="pf-c-card__header makeStyles-header-1"
        onClick={[Function]}
      >
        <div
          className="pf-c-card__header-toggle"
        >
          <button
            aria-disabled={false}
            aria-expanded={true}
            aria-label="Toggle details"
            className="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-3"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={false}
            id="toggle-button"
            onClick={[Function]}
            role={null}
            type="button"
          >
            <span
              className="pf-c-card__header-toggle-icon"
            >
              <svg
                aria-hidden="true"
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="pf-c-card__title"
        >
          Learn more about the terminology
        </div>
      </div>
      <div
        className="pf-c-card__expandable-content makeStyles-body-2"
      >
        <div
          className="pf-c-card__body"
        >
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Subscriptions
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Subscriptions identify Kubernetes resources within channels (source repositories). Then, the subscription places the Kubernetes resources on the target clusters.
            </p>
          </div>
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Placement rules
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Placement rules define the target clusters where subscriptions are delivered. This is done by cluster name, cluster resource annotation(s), or cluster resource label(s).
            </p>
          </div>
          <div
            className="terminology-item"
          >
            <p
              className="deployment-highlights-terminology-title"
            >
              Channels
            </p>
            <p
              className="deployment-highlights-terminology-content"
            >
              Channels point to repositories where Kubernetes resources are stored, such as Git, Helm chart, or object storage repositories, or Namespaces on the local cluster. Channels support multiple subscriptions from multiple targets.
            </p>
          </div>
          <div
            className="deployment-highlights-terminology-docs"
          >
            <a
              href="https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.4/html/applications/managing-applications"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="deployment-highlights-terminology-docs-text"
              >
                View documentation
              </span>
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                className="details-item-link-icon"
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </article>
  </div>
</div>
`;
